be/src/exec/pipeline/adaptive/collect_stats_source_operator.cpp

```diff
@@ -73,6 +73,8 @@ void CollectStatsSourceOperatorFactory::close(RuntimeState* state) {
 }
 
 OperatorPtr CollectStatsSourceOperatorFactory::create(int32_t degree_of_parallelism, int32_t driver_sequence) {
+    // DOP will be recomputed when its group is ready, see CollectStatsSourceOperatorFactory::degree_of_parallelism.
+    _ctx->set_downstream_dop(degree_of_parallelism);
```

Comment:
both SourceOperator and SinkOperator can modify one another's dop,  can this dop modification keep unilateral?

Reply:
NO, only `SinkOperator` modify the DOP of `SourceOperator`.
downstream_dop is source dop, and upstream dop is sink dop.

Reply:
Delete the methods `set_upstream_dop` and `set_downstream_dop`.